libraries/postgres/src/Database.js

```diff
@@ -221,6 +222,51 @@ class Database {
         }
       }
 
+      // look for unexpected user attributes
+      const badAttrs = {
+        'superuser': 'rolsuper',
+        'createrole': 'rolcreaterole',
+        'createdb': 'rolcreatedb',
+        'replication': 'rolreplication',
+      };
```

Comment:
just out of curiousity, why do we do `badAttrs` rather than confirm that we only have `goodAttrs`. Do we want to allow for arbitrary roles in general?

Reply:
These are separate from roles.  These are different columns, in a join, so *conceivably* we could use `select *` to get all of the columns available, and then fail if there are any unknown columns with a `true` value.  I don't know if such columns would be added in a major or minor version of pg, though.  It runs the risk of spurious failures if pg adds a new benign attribute like `login`.  I'm not sure it's worth painting all the way into that corner..